src/ServicePulse.Host/vue/src/components/failedmessages/FlowDiagram.vue

```diff
@@ -1,48 +1,70 @@
-import { useFetchFromServiceControl } from "../../composables/serviceServiceControlUrls";
-import { MarkerType, useVueFlow, VueFlow } from "@vue-flow/core";
+import { useTypedFetchFromServiceControl } from "../../composables/serviceServiceControlUrls";
+import { type DefaultEdge, MarkerType, useVueFlow, VueFlow, type Styles, type Node } from "@vue-flow/core";
 import TimeSince from "../TimeSince.vue";
 import routeLinks from "@/router/routeLinks";
-
-const props = defineProps({
-  conversationId: String,
-  messageId: String,
-});
-
-const messageType = Object.freeze({
-  Event: "Event message",
-  Timeout: "Timeout message",
-  Command: "Command message",
-});
+import Message from "@/resources/Message";
+
+const props = defineProps<{
+  conversationId?: string;
+  messageId: string;
+}>();
+
+enum MessageType {
+  Event = "Event message",
+  Timeout = "Timeout message",
+  Command = "Command message",
+}
+
+interface MappedMessage {
+  nodeName: string;
+  id: string;
+  messageId: string;
+  sendingEndpoint: string;
+  receivingEndpoint: string;
+  parentId: string;
+  parentEndpoint: string;
+  type: MessageType;
+  isError: boolean;
+  sagaName: string;
+  link: {
+    name: string;
+    nodeName: string;
+  };
+  timeSent: string;
+  level?: number;
+  width?: number;
+  XPos?: number;
+}
 
 const nodeSpacingX = 300;
 const nodeSpacingY = 200;
 
-async function getConversation(conversationId) {
-  const response = await useFetchFromServiceControl(`conversations/${conversationId}`);
-  return response.json();
+async function getConversation(conversationId: string) {
+  const [, data] = await useTypedFetchFromServiceControl<Message[]>(`conversations/${conversationId}`);
+  return data;
 }
 
-function mapMessage(message) {
+function mapMessage(message: Message): MappedMessage {
   let parentId = "",
     parentEndpoint = "",
     sagaName = "";
   const header = message.headers.find((header) => header.key === "NServiceBus.RelatedTo");
   if (header) {
-    parentId = header.value;
-    parentEndpoint = message.headers.find((h) => h.key === "NServiceBus.OriginatingEndpoint")?.value;
+    parentId = header.value ?? "";
+    parentEndpoint = message.headers.find((h) => h.key === "NServiceBus.OriginatingEndpoint")?.value ?? "";
   }
 
   const sagaHeader = message.headers.find((header) => header.key === "NServiceBus.OriginatingSagaType");
   if (sagaHeader) {
-    sagaName = sagaHeader.value.split(", ")[0];
+    sagaName = sagaHeader.value?.split(", ")[0] ?? "";
   }
 
   const type = (() => {
-    if (message.headers.find((header) => header.key === "NServiceBus.MessageIntent").value === "Publish") return messageType.Event;
-    else if (message.headers.find((header) => header.key === "NServiceBus.IsSagaTimeoutMessage")?.value?.toLowerCase() === "true") return messageType.Timeout;
-    return messageType.Command;
+    if (message.headers.find((header) => header.key === "NServiceBus.MessageIntent")?.value === "Publish") return MessageType.Event;
```

Comment:
I think MessageIntent is an enum in c# land, should we have a similar enumeration in ts?

Reply:
ping

Reply:
https://github.com/Particular/ServicePulse/pull/1769/commits/8527ab283ca91408b6c6f80028064f398cdb8618